Core/NoOpRunner.Core/Iterators/Iterator.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace NoOpRunner.Core
+{
+    abstract class Iterator : IEnumerator
```

Comment:
I don't think they'll like you extending `IEnumerator` for this

Reply:
https://refactoring.guru/design-patterns/iterator/csharp/example